Create Inventory User association with user_roles

Exellin opened the pull request:
read, write, and admin levels through an enumerator
created inventory_users factory with the 3 roles as traits
tested inventory assocation in user_spec
tested user assocaition in inventory_spec

resolves: #31, #29

pachonk commented on .rubocop_todo.yml:
```diff
@@ -1,6 +1,6 @@
 # This configuration was generated by
```
This file really shouldn't be touched unless there's a deletion after initial config. Why was it changed?

Exellin replied:
We haven't talked too much about what rules we are going ignore or what styles we want to add to our models and classes etc to adhere to rubocop standards

I had the following errors before re-running the configuration file
spec/models/inventory_spec.rb:3:1: C: Block has too many lines. [45/35]
spec/models/user_spec.rb:3:1: C: Block has too many lines. [37/35]
app/models/inventory_user.rb:1:1: C: Missing top-level class documentation comment.

If you have some suggestions in how I can properly add another block to the tests I will gladly implement it but don't have any idea at the moment

Do we want to add a comment documenting what each class is for each ruby file?  If so I can start here.

Otherwise, I just added these offenses into the rubocop file since I didn't want to have this pull request fail because of rules/standards we have't decided on yet.

Is it better practice to bring in a pull request with failing rubocop tests that will be pulled in anyway, or how should I handle this in the future?

pachonk replied:
TL;DR, busy, will read later.
Anything we don't want to enforce should go into the .rubocop.yml

Exellin replied:
Do we want to enforce any of the above rules then or should I put them into .rubocop.yml?

pachonk replied:
Let's blacklist spec for too many lines.
Let's do research on whether or not we want to enforce top level class comments as a team.

Exellin replied:
I'm of the opinion that top level comments aren't really necessary for a rails projects unless you are adding a lot of custom functionality.  You know a controller is going to list available actions and fill in the logic for each action, and a model is going to list the relationships, validations, and custom models.

But I haven't really worked on a complicated rails projects and don't know how complicated classes can get and whether comments explaining the functionality of a class is a good habit to get in to even though it might feel useless to start off with.

pachonk commented on app/models/inventory_user.rb:
```diff
@@ -0,0 +1,12 @@
+  belongs_to :user
+  belongs_to :inventory
+
+  enum user_role: {
```
Isn't this really more of an inventory role than a user role?

Exellin replied:
It's a user's role on the inventory.  I think inventory role implies the role of the inventory (something like transport, sell, etc).

pachonk replied:
Can we use symbols instead of strings? `:read, :write, :admin`

Exellin replied:
@pachonk Since you already merged this I guess we can change this when we tackle issue #17

pachonk commented on spec/models/user_spec.rb:
```diff
@@ -26,4 +26,23 @@
+
+    it 'is valid when it has inventories' do
+      inventories = FactoryGirl.create_list(:inventory, 2)
+      (0..inventories.length - 1).each do |i|
```
Why not just do `inventories.each do |inventory|`?

Exellin replied:
Just didn't think of it at the time.  Fixed for both files.

pachonk commented on spec/models/inventory_spec.rb:
```diff
@@ -37,4 +37,23 @@
   end
+
+  describe '#user association' do
+    before do
```
Use rspec `let`

pachonk replied:
Or `subject`

Exellin replied:
done

Exellin commented:
I reverted the changes to rubocop_todo.yml and so far have the following in .rubocop.yml

Style/BlockLength:
  Exclude:
    - 'spec/**/*.rb'
    - Guardfile

then I also deleted the corresponding BlockLength in the todo file.

I also commented earlier on the comments for classes, what are your thoughts @pachonk?

pachonk commented on app/models/inventory.rb:
```diff
@@ -1,4 +1,7 @@
   validates :name, presence: true, length: { maximum: 255 }, uniqueness: { case_sensitive: false }
   validates :description, length: { maximum: 255 }
+
+  has_many :inventory_users
```
Also need a has_one owner (of user or org)

Exellin replied:
Yeah I assumed that was going to be implemented with #79

pachonk replied:
Sounds good :ok_hand:

pachonk commented on app/models/organization.rb:
```diff
@@ -1,5 +1,6 @@
 class Organization < ApplicationRecord
   validates :name, presence: true, uniqueness: { case_sensitive: false}, length: { maximum: 255 }
+
```
Usually if you're not actually editing this file you'd want to leave an edit like this out for means of a more concise diff.

Exellin replied:
@pachonk Would I make a separate PR for just this change, or leave it as is until I make a PR which changes this file?
